### ISSUE TITLE
Move Cloud Hypervisor to virtio-vsock crate from rust-vmm

### ISSUE THREAD

**Issue by sboeuf** [cloud-hypervisor/cloud-hypervisor#3822]
In order to share the maximum amount of code with the upstream regarding the `virtio-vsock` device, it'd be good to move the codebase to the upstream crate https://github.com/rust-vmm/vm-virtio/tree/main/crates/devices/virtio-vsock

**Comment by andreeaflorescu** [comment]
We still have to upstream a few things, so far we only got the packet implementation. The other ones will follow soon.

**Comment by sboeuf** [comment]
Yes of course, but I wanted to start tracking this :)

**Comment by sboeuf** [comment]
@andreeaflorescu is someone actively working on the `virtio-vsock` crate to make it ready to be pulled by external projects?

**Comment by andreeaflorescu** [comment]
@sboeuf we already published the vsock-packet sub-component from virtio-vsock. The other ones will follow in the next couple of months. We want to finish the upstreaming for them by the end of 2022. If you'd like you can consume vsock-packet already, otherwise you can also wait for the other ones to be published.

**Comment by lkml-likexu** [comment]
Hi @sboeuf @andreeaflorescu 

I noticed that the Virtio-VSOCK feature stops working if a v6.3+ guest kernel is booted with the v33.0 CH.

This is due to the kernel commit [71dc9ec9ac7d](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=71dc9ec9ac7d3eee785cdc986c3daeb821381e20) ("virtio/vsock: replace virtio_vsock_pkt with sk_buff"), and CH's virtio-vsock backend crate doesn't have no support for parsing the new struct sk_buff. In that case, the endless WARN log "WARN --- <vsock> virtio-devices/src/vsock/device.rs:146 -- vsock: RX queue error: BufDescMissing" is easily triggered with documented steps in the [docs/vsock.md](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/vsock.md). 

I'd like to know your thoughts on addressing this bug, an important issue here is how to get the CH backend to maintain compatibility in parsing different formats for guests with unspecified kernel versions.

**Comment by stefano-garzarella** [comment]
Hi @lkml-likexu, it looks like the same issue we fixed in rust-vmm's vsock crate: https://github.com/rust-vmm/vm-virtio/issues/204

**Comment by rbradford** [comment]
I don't think this makes sense any more - our implementation has been hardened extensively.
